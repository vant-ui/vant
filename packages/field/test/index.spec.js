import Field from '../';
import demoTest from '../../../test/demo-test';

demoTest(Field);
