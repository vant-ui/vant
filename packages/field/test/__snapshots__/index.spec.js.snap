// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders field correctly 1`] = `
<section class="van-doc-demo-section demo-field" style="background:undefined;">
  <section class="van-doc-demo-block">
    <h2 class="van-doc-demo-block__title">基础用法</h2>
    <div class="van-cell-group van-hairline--top-bottom">
      <div placeholder="请输入用户名" class="van-cell van-hairline van-field">
        <!---->
        <!---->
        <div class="van-cell__value van-cell__value--alone">
          <input type="text" placeholder="请输入用户名" value="" class="van-field__control">
          <!---->
          <!---->
          <!---->
        </div>
        <!---->
      </div>
    </div>
  </section>
  <section class="van-doc-demo-block">
    <h2 class="van-doc-demo-block__title">自定义类型</h2>
    <div class="van-cell-group van-hairline--top-bottom">
      <div placeholder="请输入用户名" class="van-cell van-cell--required van-hairline van-field van-field--has-icon">
        <!---->
        <div class="van-cell__title"><span>用户名</span>
          <!---->
        </div>
        <div class="van-cell__value">
          <input type="text" placeholder="请输入用户名" value="" class="van-field__control">
          <!---->
          <div class="van-field__icon" style="display:none;">
            <i class="van-icon van-icon-clear">
              <!---->
            </i>
          </div>
          <!---->
        </div>
        <!---->
      </div>
      <div placeholder="请输入密码" class="van-cell van-cell--required van-hairline van-field">
        <!---->
        <div class="van-cell__title"><span>密码</span>
          <!---->
        </div>
        <div class="van-cell__value">
          <input type="password" placeholder="请输入密码" value="" class="van-field__control">
          <!---->
          <!---->
          <!---->
        </div>
        <!---->
      </div>
    </div>
  </section>
  <section class="van-doc-demo-block">
    <h2 class="van-doc-demo-block__title">禁用输入框</h2>
    <div class="van-cell-group van-hairline--top-bottom">
      <div disabled="disabled" class="van-cell van-hairline van-field">
        <!---->
        <div class="van-cell__title"><span>用户名</span>
          <!---->
        </div>
        <div class="van-cell__value">
          <input type="text" disabled="disabled" value="输入框已禁用" class="van-field__control">
          <!---->
          <!---->
          <!---->
        </div>
        <!---->
      </div>
    </div>
  </section>
  <section class="van-doc-demo-block">
    <h2 class="van-doc-demo-block__title">错误提示</h2>
    <div class="van-cell-group van-hairline--top-bottom">
      <div placeholder="请输入用户名" class="van-cell van-hairline van-field van-field--error">
        <!---->
        <div class="van-cell__title"><span>用户名</span>
          <!---->
        </div>
        <div class="van-cell__value">
          <input type="text" placeholder="请输入用户名" value="" class="van-field__control">
          <!---->
          <!---->
          <!---->
        </div>
        <!---->
      </div>
      <div placeholder="请输入手机号" class="van-cell van-hairline van-field">
        <!---->
        <div class="van-cell__title"><span>手机号</span>
          <!---->
        </div>
        <div class="van-cell__value">
          <input type="text" placeholder="请输入手机号" value="1365577" class="van-field__control">
          <div class="van-field__error-message">手机号格式错误</div>
          <!---->
          <!---->
        </div>
        <!---->
      </div>
    </div>
  </section>
  <section class="van-doc-demo-block">
    <h2 class="van-doc-demo-block__title">高度自适应</h2>
    <div class="van-cell-group van-hairline--top-bottom">
      <div placeholder="请输入留言" rows="1" class="van-cell van-hairline van-field">
        <!---->
        <div class="van-cell__title"><span>留言</span>
          <!---->
        </div>
        <div class="van-cell__value">
          <textarea placeholder="请输入留言" rows="1" class="van-field__control"></textarea>
          <!---->
          <!---->
          <!---->
        </div>
        <!---->
      </div>
    </div>
  </section>
  <section class="van-doc-demo-block">
    <h2 class="van-doc-demo-block__title">插入按钮</h2>
    <div class="van-cell-group van-hairline--top-bottom">
      <div placeholder="请输入短信验证码" class="van-cell van-cell--center van-hairline van-field van-field--has-icon">
        <!---->
        <div class="van-cell__title"><span>短信验证码</span>
          <!---->
        </div>
        <div class="van-cell__value">
          <input type="text" placeholder="请输入短信验证码" value="" class="van-field__control">
          <!---->
          <div class="van-field__icon" style="display:none;">
            <i class="van-icon van-icon-clear">
              <!---->
            </i>
          </div>
        </div>
        <!---->
        <div class="van-field__button">
          <button class="van-button van-button--primary van-button--small">
            <!----><span class="van-button__text">发送验证码</span></button>
        </div>
      </div>
    </div>
  </section>
</section>
`;
